ntbs-service/Pages/Notifications/Edit/Table/ManualTestResult.cshtml.cs

```diff
@@ -0,0 +1,132 @@
+                }
+                else
+                {
+                    await _testResultsRepository.UpdateTestResultAsync(TestResultForEdit);
+                }
+            }
+        }
+
+        protected override async Task<IActionResult> PrepareAndDisplayPageAsync(bool isBeingSubmitted)
+        {
+            if (RowId != null)
+            {
+                TestResultForEdit = Notification.TestData.ManualTestResults
+                    .SingleOrDefault(r => r.ManualTestResultId == RowId.Value);
+                if (TestResultForEdit != null)
+                {
+                    FormattedTestDate = TestResultForEdit.TestDate.ConvertToFormattedDate();
+                }
+            }
+
+            await SetDropdownsAsync();
+
+            return Page();
+        }
+
+        private async Task SetDropdownsAsync()
+        {
+            var manualTestTypes = await _referenceDataRepository.GetManualTestTypesAsync();
+            ManualTestTypes = new SelectList(manualTestTypes, nameof(ManualTestType.ManualTestTypeId), nameof(ManualTestType.Description));
+
+            var sampleTypes = await _referenceDataRepository.GetSampleTypesAsync();
+            SampleTypes = new SelectList(
+                items: sampleTypes,
+                dataValueField: nameof(SampleType.SampleTypeId),
+                dataTextField: nameof(SampleType.Description),
+                selectedValue: null,
+                dataGroupField: nameof(SampleType.Category));
+
+            ResultOptions = new SelectList(ResultEnumHelper.GetAll());
+        }
+
+        protected override IActionResult RedirectAfterSaveForNotified()
+        {
+            return RedirectToPage("/Notifications/Edit/TestResults", new { NotificationId });
+        }
+
+        protected override IActionResult RedirectAfterSaveForDraft(bool isBeingSubmitted)
+        {
+            return RedirectToPage("/Notifications/Edit/TestResults", new { NotificationId, isBeingSubmitted });
+        }
+
+        protected override async Task<Notification> GetNotificationAsync(int notificationId)
+        {
+            return await NotificationRepository.GetNotificationWithTestsAsync(notificationId);
+        }
+
+        private async Task SetRelatedEntities()
+        {
+            if (TestResultForEdit.ManualTestTypeId != null)
+            {
+                TestResultForEdit.ManualTestType = await _referenceDataRepository.GetManualTestTypeAsync(TestResultForEdit.ManualTestTypeId.Value);
+            }
+            if (TestResultForEdit.SampleTypeId != null)
+            {
+                TestResultForEdit.SampleType = await _referenceDataRepository.GetSampleTypeAsync(TestResultForEdit.SampleTypeId.Value);
+            }
+        }
+
+        private void SetDate()
+        {
+            // The required date will be marked as missing on the model, since we are setting it manually, rather than binding it
+            ModelState.Remove("TestResultForEdit.TestDate");
```

Comment:
Do we have to do this elsewhere too where we set dates? Or is this case slightly different?

Reply:
I have checked other places for this and it didn't seem problematic. I think it is different in that we don't have require other dates, at best with `RequiredIf("ShouldValidateFull")`